src/context/favorites/FavoritesProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import { useLocalStorage } from 'hooks/useLocalStorage';
+import { createContext, useContext, PropsWithChildren, useReducer, useEffect } from 'react';
+import { favoritesReducer } from './FavoritesReducer';
+import { State, Dispatch } from './models';
+
+interface ContextProps {
+  state: State;
+  dispatch: Dispatch;
+  checkIfIsFavorite: (city: string) => boolean;
+}
+
+const initialState: State = {
+  favorites: [],
+};
+
+const FavoriteContext = createContext<ContextProps>({
+  dispatch: () => {},
+  state: initialState,
+  checkIfIsFavorite: () => false,
+});
+
+const FavoritesProvider = ({ children }: PropsWithChildren) => {
+  const [favorites, setFavorites] = useLocalStorage<string[]>('favorites');
+  const [state, dispatch] = useReducer(favoritesReducer, { favorites: favorites || [] });
+
+  const checkIfIsFavorite = (city: string): boolean => {
+    return state.favorites.includes(city);
+  };
+
+  useEffect(() => {
+    setFavorites(state.favorites);
```

Comment:
**question** i dont see serialization array to string, how it is handled?

Reply:
Hook useLocalStorage handles JSON.stringify and JSON.parse before and after localStorage functions